platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/fan_drawer.py

```diff
@@ -96,5 +107,16 @@ def get_maximum_consumed_power(self):
         """
         return 36.0
 
+def get_platform_name():
+    return _product_dict.get(device_info.get_platform())
+
 def fan_drawer_list_get():
-    return [FanDrawer(0)]
+    global  _MAX_FANTRAY, _MAX_FAN
+    platform = get_platform_name()
+    if platform == "Newport":
+        _MAX_FANTRAY = 1
+        _MAX_FAN = 6
+    if platform == "Mavericks":
+        _MAX_FANTRAY = 2
+        _MAX_FAN = 5
+    return [FanDrawer(i, platform) for i in range(0, _MAX_FANTRAY)]
```

Comment:
FanDrawer index starts from 0
Fan index starts from 1
Please make it the same (start from 1)